nat_gateway/output.tf

```diff
@@ -1,7 +1,8 @@
 output "ids" {
-  value = ["${aws_nat_gateway.gateway.*.id}"]
+  value = [aws_nat_gateway.gateway.*.id]
```

Comment:
you don't need the list notation here, as `aws_nat_gateway.gateway.*.id` is already a list